### PR TITLE
80 implement rest get route for weekly action items

### LINKED ISSUE #80: Implement REST GET route for weekly action items
## Description
The weekly action items are stored in the DB but are not accessible through REST API via the backend

## DoD
A GET route for action-items returns the business weekly action items

### PR DESCRIPTION
**Pull request by liadyacobi**

### CODE REVIEW

File: backend/src/controllers/action-items.controller.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  constructor() {
+    super(WeeklyActionItemsModel);
+  }
+
+  async getWeeklyActionItems(req: AuthRequest, res: Response<IWeeklyActionItems>) {
+    const { businessId } = req.user!;
+    const [weeklyActionItems] = await WeeklyActionItemsModel.find({ businessId }).sort({ date: -1 }).limit(1);
+
+    if (!weeklyActionItems) {
+      return res.status(httpStatus.NOT_FOUND).send();
```
Comment (liadyacobi) [inline backend/src/controllers/action-items.controller.ts]: Return a positive status indicating no action items

File: backend/src/models/weekly-action-items.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import mongoose from 'mongoose';
+import { IWeeklyActionItems, WeeklyActionItemsMongooseSchema } from 'shared-types';
+
+const WeeklyActionItemsSchema = new mongoose.Schema<IWeeklyActionItems>(WeeklyActionItemsMongooseSchema.schema);
+
+WeeklyActionItemsSchema.index({ businessId: 1, date: 1 });
```
Comment (chenyair) [inline backend/src/models/weekly-action-items.ts]: What is this? Index it in the WeeklyActionItemsMongooseSchema. You can add index: true to the required fields to be indexed

File: backend/src/controllers/action-items.controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import httpStatus from 'http-status';
+
+class ActionItemsController extends BaseController<IWeeklyActionItems> {
+  constructor() {
+    super(WeeklyActionItemsModel);
+  }
+
+  async getWeeklyActionItems(req: AuthRequest, res: Response<IWeeklyActionItems>) {
+    const { businessId } = req.user!;
+    const [weeklyActionItems] = await WeeklyActionItemsModel.find({ businessId }).sort({ date: -1 }).limit(1);
```
Comment (chenyair) [inline backend/src/controllers/action-items.controller.ts]: Use findOne

**Review by chenyair — CHANGES_REQUESTED** [review]

File: backend/src/controllers/action-items.controller.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import httpStatus from 'http-status';
+
+class ActionItemsController extends BaseController<IWeeklyActionItems> {
+  constructor() {
+    super(WeeklyActionItemsModel);
+  }
+
+  async getWeeklyActionItems(req: AuthRequest, res: Response<IWeeklyActionItems>) {
+    const { businessId } = req.user!;
+    const weeklyActionItems = await WeeklyActionItemsModel.findOne({ businessId }).sort({ date: -1 }).limit(1);
```
Comment (yoavrichter30) [inline backend/src/controllers/action-items.controller.ts]: Why findOne + sort + limit 1?

**Review by yoavrichter30 — CHANGES_REQUESTED** [review]

**Review by yoavrichter30 — APPROVED** [review]

**Pull request merged**